_posts/2019-08-24-inception-pipelines-pt6.markdown

```diff
@@ -0,0 +1,148 @@
+---
+layout: post
+title:  "Seeds of Inception - Part 6"
+date:   2018-08-24
+tags: aws codepipeline inception pipeline iam roles
+author: Pete Yandell
+image: img/inception-pipelines/seed_germination.png
+---
+
+**Part 6 planting the seed** - A practical example for DevOps on AWS
+
+![vines]({{ site.url }}/img/inception-pipelines/part-6-planting-the-seed.jpg)
+
+## What's The Problem
+
+In this month's instalment I'm going right back to the beginning to answer a question that I'm frequently asked; namely *"How do I do the initial seeding of a pipeline in my shiny new AWS Account?"*.
+
+Like all things AWS-ey, there are multiple ways to accomplish this. The three that immediately come to mind are:
+
+1. Use the AWS Account root user credentials
+2. Create an IAM user directly in the AWS Account .
+3. Assuming the /roles/OrganizationAccountAccessRole role that was created if your AWS Account was created using AWS Organisations.
+
+The more versatile and secure way is **Option 2**, which I'll discuss in detail below. My (personal) reasoning for this is you only need to keep the credentials around for as long as it takes to seed the pipeline and add more specific accounts as documented in [Part 5](https://mechanicalrock.github.io/2018/07/31/inception-pipelines-pt5.html). Once you've done this, delete the credentials to make the IAM User secure. If/when you need to seed another pipeline, simply regenerate new credentials and run from Step 4 again.
+
+**Warning:** This post gets a bit heavy on the screenshots in a couple of steps. So, brace yourself and accept my apologies in advance!
+
+Let's dig in!
+
+## Step 1 - AWS Account
+
+The very first step is making sure you have an AWS Account and you have either root, admin or enough permissions to create an IAM user and policy. Oh, you also need to log into the [AWS Console](https://console.aws.amazon.com/console/home).
+
+## Step 2 - Create A Customer Managed IAM policy
+
+Follow the four simple actions below to create a [Customer Managed IAM Policy](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_managed-vs-inline.html#customer-managed-policies). You'll use this policy in the next step when we create an [IAM User](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users.html).
```

Comment:
Customer Managed IAM Policy and the code pipeline user should be a CloudFormation template.  

Getting access keys can be a manual step.

Reply:
I agree in principal, however wouldn't then be another manual step to run the CFN template and then another to get the credentials. CFN templates should always be managed via InceptionPipelines :)